pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -1095,6 +1051,54 @@ func (c *K8sOrchestrator) GetAllK8sVolumes() []string {
 	return volumeIDs
 }
 
+// SetWcpCapabilitiesMap reads the capabilities values from 'supervisor-capabilities' CR in
+// supervisor cluster and sets values in global wcp capabilities map.
+func SetWcpCapabilitiesMap(ctx context.Context, wcpCapabilityApiClient client.Client) error {
+	log := logger.GetLogger(ctx)
+	// Check the 'supervisor-capabilities' CR in supervisor
+	wcpCapabilities := &wcpcapv1alph1.Capabilities{}
+	err := wcpCapabilityApiClient.Get(ctx, k8stypes.NamespacedName{
+		Name: common.WCPCapabilitiesCRName},
+		wcpCapabilities)
+	if err != nil {
+		log.Errorf("failed to fetch Capabilities CR instance "+
+			" with name %q Error: %+v", common.WCPCapabilitiesCRName, err)
+		return err
+	}
+
+	wcpCapabilitiesMapMutex.Lock()
+	defer wcpCapabilitiesMapMutex.Unlock()
+	if WcpCapabilitiesMap == nil {
+		WcpCapabilitiesMap = make(map[string]bool)
+	}
+	for capName, capStatus := range wcpCapabilities.Status.Supervisor {
+		WcpCapabilitiesMap[string(capName)] = capStatus.Activated
+	}
+	return nil
+}
+
+// setCapabilitiesMapOnSupervisor creates a new cns-operator client for supervisor cluster and
```

Comment:
can we reword this as "new api client for supervisor capability" instead of "new cns-operator client" ?